src/main/java/org/cactoos/iterable/Solid.java

```diff
@@ -46,8 +46,8 @@ public Solid(final X... src) {
      * Ctor.
      * @param iterable The iterable
      */
-    public Solid(final Iterable<X> iterable) {
-        super(new Synced<>(new Sticky<>(iterable)));
+    public Solid(final Iterable<? extends X> iterable) {
+        super(new Synced<>(new Sticky<>(new Mapped<>(x -> x, iterable))));
```

Comment:
@andreoss here we have mapped, we could either do casting as discussed above or add a todo to find a better solution. But I would prefer that if we choose one way, we apply the same everywhere though.

Reply:
@victornoel Here it was unnecessary. Removed it